.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,46 @@
+name: GitHub Pages
+
+on:
+  push:
+    branches:
+      - main  # Set a branch to deploy
+  pull_request:
```

Comment:
Do we want to deploy the docs on PR's to `main`?

Reply:
We don't. But I think the `${{ github.ref == 'refs/heads/main' }}` is not met on such PR's, thus preventing the actual deployment.